packages/discord.js/src/structures/interfaces/Collector.js

```diff
@@ -96,21 +102,33 @@ class Collector extends EventEmitter {
    * @emits Collector#collect
    */
   async handleCollect(...args) {
-    const collect = await this.collect(...args);
-
-    if (collect && (await this.filter(...args, this.collected))) {
-      this.collected.set(collect, args[0]);
-
-      /**
-       * Emitted whenever an element is collected.
-       * @event Collector#collect
-       * @param {...*} args The arguments emitted by the listener
-       */
-      this.emit('collect', ...args);
-
-      if (this._idletimeout) {
-        clearTimeout(this._idletimeout);
-        this._idletimeout = setTimeout(() => this.stop('idle'), this.options.idle).unref();
+    const collectedId = await this.collect(...args);
+    const filterResult = await this.filter(...args, this.collected);
+
+    if (collectedId) {
```

Comment:
```suggestion

    if (collectedId) {
        const filterResult = await this.filter(...args, this.collected);
```
the filter should only be called for items that pass the collect check